simapp/app_test.go

```diff
@@ -199,36 +208,85 @@ func TestRunMigrations(t *testing.T) {
+	mockDefaultGenesis := json.RawMessage(`{"key": "value"}`)
+	mockModule.EXPECT().DefaultGenesis(gomock.Eq(app.appCodec)).Times(1).Return(mockDefaultGenesis)
+	mockModule.EXPECT().InitGenesis(gomock.Eq(ctx), gomock.Eq(app.appCodec), gomock.Eq(mockDefaultGenesis)).Times(1).Return(nil)
+	mockModule.EXPECT().ConsensusVersion().Times(1).Return(uint64(0))
+
+	app.mm.Modules["mock"] = mockModule
+
+	// Run migrations only for "mock" module. We exclude it from
+	// the VersionMap to simulate upgrading with a new module.
+	_, err := app.mm.RunMigrations(ctx, app.Configurator(),
+		module.VersionMap{
+			"bank":            bank.AppModule{}.ConsensusVersion(),
+			"auth":            auth.AppModule{}.ConsensusVersion(),
+			"authz":           authzmodule.AppModule{}.ConsensusVersion(),
+			"staking":         staking.AppModule{}.ConsensusVersion(),
+			"mint":            mint.AppModule{}.ConsensusVersion(),
+			"distribution":    distribution.AppModule{}.ConsensusVersion(),
+			"slashing":        slashing.AppModule{}.ConsensusVersion(),
+			"gov":             gov.AppModule{}.ConsensusVersion(),
+			"params":          params.AppModule{}.ConsensusVersion(),
+			"upgrade":         upgrade.AppModule{}.ConsensusVersion(),
+			"vesting":         vesting.AppModule{}.ConsensusVersion(),
+			"feegrant":        feegrantmodule.AppModule{}.ConsensusVersion(),
+			"evidence":        evidence.AppModule{}.ConsensusVersion(),
+			"crisis":          crisis.AppModule{}.ConsensusVersion(),
+			"genutil":         genutil.AppModule{}.ConsensusVersion(),
+			"capability":      capability.AppModule{}.ConsensusVersion(),
+			"ibc":             ibc.AppModule{}.ConsensusVersion(),
+			"epochs":          epochs.AppModule{}.ConsensusVersion(),
+			"halving":         halving.AppModule{}.ConsensusVersion(),
+			"interchainquery": interchainquery.AppModule{}.ConsensusVersion(),
+			"ibchooker":       ibchooker.AppModule{}.ConsensusVersion(),
+			"oracle":          oracle.AppModule{}.ConsensusVersion(),
+		},
+	)
+	require.NoError(t, err)
+}
+
 func TestUpgradeStateOnGenesis(t *testing.T) {
-	encCfg := MakeTestEncodingConfig()
 	db := dbm.NewMemDB()
 	logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
 	app := NewSimappWithCustomOptions(t, false, SetupOptions{
-		Logger:             logger,
-		DB:                 db,
-		InvCheckPeriod:     0,
-		EncConfig:          encCfg,
-		HomePath:           DefaultNodeHome,
-		SkipUpgradeHeights: map[int64]bool{},
-		AppOpts:            EmptyAppOptions{},
+		Logger:  logger,
+		DB:      db,
+		AppOpts: simtestutil.NewAppOptionsWithFlagHome(t.TempDir()),
 	})
 
 	// make sure the upgrade keeper has version map in state
 	ctx := app.NewContext(false, tmproto.Header{})
 	vm := app.UpgradeKeeper.GetModuleVersionMap(ctx)
-
 	for v, i := range app.mm.Modules {
-		require.Equal(t, vm[v], i.ConsensusVersion())
+		if i, ok := i.(module.HasConsensusVersion); ok {
+			require.Equal(t, vm[v], i.ConsensusVersion())
```

Comment:
why this line is used 3 times?

Reply:
Not sure how did I end up with 3 times 🤔 
Changing it to one time.